exporter/signalfxexporter/translation/translator.go

```diff
@@ -87,19 +188,66 @@ func createDimensionsMap(rules []Rule) map[string]string {
 	return nil
 }
 
-func (mp *MetricTranslator) TranslateDataPoints(sfxDataPoints []*sfxpb.DataPoint) {
+func (mp *MetricTranslator) TranslateDataPoints(logger *zap.Logger, sfxDataPoints []*sfxpb.DataPoint) []*sfxpb.DataPoint {
+	processedDataPoints := sfxDataPoints
+
 	for _, tr := range mp.rules {
-		for _, dp := range sfxDataPoints {
+		newDataPoints := []*sfxpb.DataPoint{}
+
+		for _, dp := range processedDataPoints {
 			switch tr.Action {
 			case ActionRenameDimensionKeys:
 				for _, d := range dp.Dimensions {
 					if newKey, ok := tr.Mapping[d.Key]; ok {
 						d.Key = newKey
 					}
 				}
+			case ActionRenameMetrics:
+				if newKey, ok := tr.Mapping[dp.Metric]; ok {
+					dp.Metric = newKey
+				}
+			case ActionMultiplyInt:
+				if multiplier, ok := tr.ScaleFactorsInt[dp.Metric]; ok {
+					v := dp.GetValue().IntValue
```

Comment:
What happens if the data point is not integer? Is IntValue==nil in that case?

Reply:
The datapoint is passed forward as is, unmodified